### PR TITLE
Api endpoint for all cards

### PR DESCRIPTION
**Pull request by DDKatch**
out

### CODE REVIEW

**Comment by DDKatch** [comment]
I found that we write in database while tests run, so @datarockets/cuties can you advice how to resolve this? What to choose? DB cleaner or mock db calls (what to use for that?)? Let's try something during PP session 😉

**Comment by AleksSenkou** [comment]
>  found that we write in database while tests run, so @datarockets/cuties can you advice how to resolve this? What to choose? DB cleaner or mock db calls (what to use for that?)? Let's try something during PP session

@datarockets/cuties `Dredd` uses real quiring too, going to create an infra task for that

File: server/spec/controllers/cards.spec.js [hunk 1]
```diff
@@ -1,28 +1,42 @@
...
+    request({
+      url: endpoint,
+      method: 'GET',
+      json: true,
+    }, (error, response) => {
+      expect(response.statusCode).toEqual(200)
+      done()
+    })
+  })
+})
```
Comment (AleksSenkou) [inline server/spec/controllers/cards.spec.js]: This file is removed in `master`. You may want to test via adding API docs for the new endpoint 😉

File: server/src/config/database.js [hunk 2]
```diff
@@ -11,10 +11,9 @@ module.exports = {
     dialect: 'postgres',
   },
   test: {
-    username: 'root',
-    password: null,
```
Comment (AleksSenkou) [inline server/src/config/database.js]: 👍

File: server/src/controllers/v1/cards.js [hunk 3]
```diff
@@ -1,12 +1,18 @@
 const { Card } = require('../../models')
 
 module.exports = {
-  create(req, res) {
-    return Card
+  create: (req, res) => {
+    Card
       .create({
         description: req.body.description,
+        raw: true,
```
Comment (AleksSenkou) [inline server/src/controllers/v1/cards.js]: @DDKatch why is it needed?

File: server/src/controllers/v1/cards.js [hunk 4]
```diff
@@ -1,12 +1,18 @@
...
       })
-      .then((card) => { res.status(201).send(card.dataValues) })
+      .then((card) => { res.status(201).send(card) })
+      .catch((error) => { res.status(400).send(error) })
+  },
+  ded: (req, res) => {
```
Comment (AleksSenkou) [inline server/src/controllers/v1/cards.js]: Guess it should be `getAll` 😄

**Review by AleksSenkou — CHANGES_REQUESTED** [review]

File: server/src/controllers/v1/cards.js (hunk 3 above)
Comment (DDKatch) [reply server/src/controllers/v1/cards.js]: @AleksSenkou in order to get array, in other way you should use `cards.dataSources` instead of just cards.

**Comment by AleksSenkou** [comment]
@DDKatch push latest changes, please. I'll finish the PR 😉

**Comment by DDKatch** [comment]
@AleksSenkou I've pushed 😉

**Comment by DDKatch** [comment]
and @AleksSenkou I've got an error after restarting test. Dredd has some similar issues about that, but they're not resolved.
screen shot 2018-12-10 at 10 59 18 am
So I think it's better to run server for tests separately.

**Comment by AleksSenkou** [comment]
@DDKatch it's already uses separate server for testing on 8000 port. Yeah, we have a problem with killing process after tests

**Comment by DDKatch** [comment]
@AleksSenkou I mean run test server not through dredd, but manually yourself. In this case there's no error.

**Comment by AleksSenkou** [comment]
@DDKatch yeah, but running server by hands isn't comfortable. We need to think about it more

**Review by AleksSenkou — APPROVED** [review]
LGTM

**Comment by nikitasakov** [comment]
👍

**Pull request merged**